x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java

```diff
@@ -206,23 +227,17 @@ protected static void checkCapabilities(RestClient client, TestFeatureService te
                 throw e;
             }
         }
-
-        var features = Stream.concat(
-            new EsqlFeatures().getFeatures().stream(),
-            new EsqlFeatures().getHistoricalFeatures().keySet().stream()
-        ).map(NodeFeature::id).collect(Collectors.toSet());
-
-        for (String feature : testCase.requiredCapabilities) {
-            var esqlFeature = "esql." + feature;
-            assumeTrue("Requested capability " + feature + " is an ESQL cluster feature", features.contains(esqlFeature));
-            assumeTrue("Test " + testName + " requires " + feature, testFeatureService.clusterHasFeature(esqlFeature));
-        }
+        return false;
     }
 
     protected boolean supportsInferenceTestService() {
         return true;
     }
 
+    protected boolean supportsIndexModeLookup() throws IOException {
```

Comment:
Imo, support for these special situations (inference and lookup index mode... so far) are a bit messy and we could improve this (in the future when maybe someone will get frustrated enough when managing an increasing list of exceptions) in the form of a `record` that has all these `boolean`s inside and instead of dealing with separate `boolean`s we deal with a single, grouped set of settings.